### PR TITLE
Add logos_check.py to validate logos filesize/resolution

### PR DESCRIPTION
**Pull request by Salamandar**
See https://github.com/YunoHost/issues/issues/2438

### CODE REVIEW

File: tools/logos_check.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+class ImageCheck():
+    def __init__(self, imgpath: Path) -> None:
+        self.imgpath = imgpath
+        self.image = Image.open(imgpath)
+
+        self.fails: list[str] = []
+
+    def check(self) -> bool:
+        self.fails.clear()
+        for fn_name in dir(self):
```
Comment (selfhoster1312) [inline tools/logos_check.py:19]: Explicitly listing the steps is clearer :)

File: tools/logos_check.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        print(f"Image '{self.imgpath}' failed tests:")
+        for fail in self.fails:
+            print(f"  - {fail}")
+        print()
+        return False
+
+    def check_type(self) -> None:
+        format = self.image.format.lower()
+        accepted_formats = ["png", "jpeg", "webp"]
```
Comment (selfhoster1312) [inline tools/logos_check.py:34]: Should we accept all those formats? Many software don't support web yet. And JPG doesn't allow background transparency which is nice for dark/light background. I think only PNG should be supported, but we could discuss it on the issue.

File: tools/logos_check.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        if format not in accepted_formats:
+            self.fails.append(f"Image should be one of {', '.join(accepted_formats)} but is {format}")
+
+    def check_dimensions(self) -> None:
+        dim_min, dim_max = 96, 300
```
Comment (selfhoster1312) [inline tools/logos_check.py:39]: Could we maybe hardcode those constants at the top of the file for easier changes?

File: tools/logos_check.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            self.fails.append(f"Dimensions should be in [{dim_min}, {dim_max}] but are {w}×{h}")
+
+    def check_ratio(self) -> None:
+        w, h = self.image.width, self.image.height
+        if w != h:
+            self.fails.append(f"Image is not square but {w}×{h}")
+
+    def check_filesize(self) -> None:
+        filesize = self.imgpath.stat().st_size
+        max_size = 80_000
```
Comment (selfhoster1312) [inline tools/logos_check.py:52]: Could we maybe hardcode this constant at the top of the file for easier changes?

File: tools/logos_check.py (hunk 2 above)
Comment (alexAubin) [reply tools/logos_check.py:34]: (currently we have `.png` exclusively in `logos/` indeed)

**Pull request closed without merging**